tests/conftest.cpp

```diff
@@ -79,14 +79,19 @@ class BaseTest: public ::testing::Test {
     test_sink->locked_backend()->add_stream(stream);
 
     // Register the sink in the logging core (BOOST_LOG)
-    logging::core::get()->add_sink(test_sink);
+    boost_logging::core::get()->add_sink(test_sink);
 
     sbuf = std::cout.rdbuf();  // save cout buffer (std::cout)
     std::cout.rdbuf(cout_buffer.rdbuf());  // redirect cout to buffer (std::cout)
 
     // todo: do this only once
     // setup a mail object
     mail::man = std::make_shared<safe::mail_raw_t>();
+
+    auto deinit_guard = logging::init(0, "test.log");
```

Comment:
What is the purpose of this logging init in "Setup" if it is immediately deinited?

Reply:
It seems to be working as expected (probably the part I'm interested in is not reset)

Forcing a failure (instead of skip), I get this output on the encoder tests.

```txt
C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Sunshine\tests\unit\test_video.cpp:30: Failure
Failed
EncoderTest(nvenc):: encoder not available


Test failed: ValidateEncoder/0

Captured boost log:
Compiling shaders...
Compiled shaders
Stereo: supported
Surround 5.1: supported
Surround 7.1: supported
Stereo: supported
Surround 5.1: supported
Surround 7.1: supported
Trying encoder [nvenc]
Failed to start ddprobe.exe:  CreateProcess failed: The system cannot find the path specified
Failed to set GPU preference. Capture may not work!

Device Description : AMD Radeon(TM) Graphics
Device Vendor ID   : 0x00001002
Device Device ID   : 0x00001638
Device Video Mem   : 495 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16048 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 1920x1080
Active GPU has HAGS disabled
Using realtime GPU priority
Desktop resolution [1920x1080]
Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
Display refresh rate [144Hz]
Requested frame rate [60fps]

Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.587891,0.364258]
Green Primary      : [0.352539,0.571289]
Blue Primary       : [0.162109,0.120117]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
Encoder [nvenc] is not supported on this GPU

Captured cout:
PlatformInitTest:: starting Fixture SetUp
PlatformInitTest:: finished Fixture SetUp
EncoderTest(nvenc):: starting Fixture SetUp
EncoderTest(nvenc):: validating encoder
PlatformInitTest:: starting Fixture TearDown
PlatformInitTest:: finished Fixture TearDown

Captured stdout:

Captured stderr:

```

Skip gets this output:

```txt
C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Sunshine\tests\unit\test_video.cpp:30: Skipped
EncoderTest(nvenc):: encoder not available


Test ignored.
```

Pass gets no output.

Reply:
Modified to keep it context through the life of the test.

Reply:
It must have something to do with `av_log_set_callback` and `av_log_set_level` then. Maybe consider putting that part of the code into a standalone function outside of `init`? It still can be called everytime in `init`, but in UTs you could "keep it clean".

Reply:
Moved the av logging setup out of the init function.